test/api_integration/apis/saved_objects/export.js

```diff
@@ -0,0 +1,363 @@
+              type: ['index-pattern', 'search', 'visualization', 'dashboard'],
+            })
+            .expect(200)
+            .then((resp) => {
+              const objects = resp.text.split('\n').map(JSON.parse);
+              expect(objects).to.have.length(3);
+              expect(objects[0]).to.have.property('id', '91200a00-9efd-11e7-acb3-3dab96693fab');
+              expect(objects[0]).to.have.property('type', 'index-pattern');
+              expect(objects[1]).to.have.property('id', 'dd7caf20-9efd-11e7-acb3-3dab96693fab');
+              expect(objects[1]).to.have.property('type', 'visualization');
+              expect(objects[2]).to.have.property('id', 'be3733a0-9efe-11e7-acb3-3dab96693fab');
+              expect(objects[2]).to.have.property('type', 'dashboard');
+            });
+        });
+
+        it('should validate types', async () => {
+          await supertest
+            .post('/api/saved_objects/_export')
+            .send({
+              type: ['foo'],
+            })
+            .expect(400)
+            .then((resp) => {
+              expect(resp.body).to.eql({
+                statusCode: 400,
+                error: 'Bad Request',
+                // eslint-disable-next-line max-len
+                message: 'child "type" fails because ["type" at position 0 fails because ["0" must be one of [index-pattern, search, visualization, dashboard]]]',
+                validation: { source: 'payload', keys: [ 'type.0' ] },
+              });
+            });
+        });
+
+        it('should validate types in objects', async () => {
+          await supertest
+            .post('/api/saved_objects/_export')
+            .send({
+              objects: [
+                {
+                  type: 'foo',
+                  id: '1',
+                },
+              ],
+            })
+            .expect(400)
+            .then((resp) => {
+              expect(resp.body).to.eql({
+                statusCode: 400,
+                error: 'Bad Request',
+                // eslint-disable-next-line max-len
+                message: 'child "objects" fails because ["objects" at position 0 fails because [child "type" fails because ["type" must be one of [index-pattern, search, visualization, dashboard]]]]',
+                validation: { source: 'payload', keys: [ 'objects.0.type' ] },
+              });
+            });
+        });
+      });
+
+      describe('10,000 objects', () => {
+        before(() => esArchiver.load('saved_objects/10k'));
+        after(() => esArchiver.unload('saved_objects/10k'));
+
+        it('should return 400 when exporting without type or objects passed in', async () => {
+          await supertest
+            .post('/api/saved_objects/_export')
+            .expect(400)
+            .then((resp) => {
+              expect(resp.body).to.eql({
+                statusCode: 400,
+                error: 'Bad Request',
+                message: '"value" must be an object',
```

Comment:
Hmm, this is a weird error. Any idea where it's coming from? I would have no idea how to fix it if that's what I received. Maybe it has do to with the `.optional()` bits added to the keys in the payload validation? `.xor()` takes care of required state, but maybe it is where the error message is so bad. I wonder if it's possible the order of `.default()` and `.xor()` matter...

Reply:
Yeah this is weird for sure, it throws the same error on any route when the body is empty. I reduced the validation to the following and causes the same error:

```
validate: {
      payload: Joi.object()
      ...
},
```